ProductsManegements/Inventory.cs

```diff
@@ -0,0 +1,198 @@
+        {
+            Product? product=SearchProduct();
+            //string pattern = @"^(?![YyNn]$).*$";
+            int input =0;
+            if (product != null)
+            {
+               void menu()
+                {
+                    Console.WriteLine(@"
+------------------------
+Choose what to modify ->
+------------------------
+");
+                    int i = 1;
+                    foreach (PropertyInfo p in product.GetType().GetProperties())
+                    {
+                        if (p.Name == "Id") continue;
+                        Console.WriteLine($"{i}.Edit {p.Name}");
+                        i++;
+                    }
+                    Console.WriteLine("0.Finish editing ");
+                }
+                while (true)
+                {
+                    menu();
+                    if (int.TryParse(Console.ReadLine(), out input))
+                    {
+                        switch (input)
+                        {
+                            case 1:
+                                string name = Console.ReadLine() ?? string.Empty;
+                                product.Name = name;
+                                Console.ForegroundColor = ConsoleColor.Green;
+                                Console.WriteLine("Product name edited successfully! ");
+                                Console.ResetColor();
+                                break;
+
+                            case 2:
+                                decimal price;
+                                do
+                                {
+                                    Console.WriteLine("Enter valid price: ");
+
+                                } while (!decimal.TryParse(Console.ReadLine(), out price));
+                                product.Price = price;
+                                Console.ForegroundColor= ConsoleColor.Green;
+                                Console.WriteLine("Price edited successfully! ");
+                                Console.ResetColor();
+
+                                break;
+
+                            case 3:
+                                int quantity;
+                                do
+                                {
+                                    Console.WriteLine("Enter the quantity of the product: ");
+
+                                } while (!int.TryParse(Console.ReadLine(), out quantity));
+                                product.Quantity = quantity;
+                                Console.ForegroundColor = ConsoleColor.Green;
+                                Console.WriteLine("Quantity edited successfully! ");
+                                Console.ResetColor();
+                                break;
+                            case 0:
+                                return;
+
+                            default:
+
+                                Console.WriteLine("Not Valid");
+                                break;
+
+
```

Comment:
0, 1, 2, and 3 are magic numbers because they are hardcoded into the logic without any named constant or comment explaining their use. So If you come back to this code after some time or maybe someone else reviews your code, it is not clear immediately what each code does. So what about using enum to represent each choice, making the code more readable?